### PR TITLE
chore: fix eth abi typings

### PR DESCRIPTION
**Pull request by chancehudson**
## Description

Copy of #4100 

@avkos @nazarhussain Can we push this through? It's been a year and this is a pretty simple type change.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-eth-abi/types/index.d.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import { AbiInput, AbiItem } from 'web3-utils';
 
-export class AbiCoder {
+declare class AbiCoder {
```
Comment (nazarhussain) [inline packages/web3-eth-abi/types/index.d.ts:22]: Removing the export would be a breaking change, that's why the tests are also failing. Please fix it, add the entry in the CHANGELOG.md file as well.

Comment (nazarhussain) [inline packages/web3-eth-abi/types/index.d.ts]: Removing the export would be a breaking change, that's why the tests are also failing. Please fix it, and add the entry in the CHANGELOG.md file as well.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: packages/web3-eth-abi/types/index.d.ts (hunk 1 above)
Comment (chancehudson) [reply packages/web3-eth-abi/types/index.d.ts]: Thanks for the heads up, I think there's no reason to change this, we just need to add a default export.

**Comment by jdevcs** [comment]
Related to https://github.com/ChainSafe/web3.js/pull/3587

**Review by avkos — APPROVED** [review]

**Review by nazarhussain — APPROVED** [review]

**Comment by avkos** [comment]
create a [copy of this PR](https://github.com/ChainSafe/web3.js/pull/4947) and close

**Pull request closed without merging**